src/ripple/app/misc/HashRouter.cpp

```diff
@@ -94,19 +93,27 @@ bool HashRouter::setFlags (uint256 const& key, int flags)
     return true;
 }
 
-bool HashRouter::swapSet (uint256 const& key, std::set<PeerShortID>& peers, int flag)
+bool HashRouter::prepRelay (uint256 const& key, std::set<PeerShortID>& peers)
 {
-    std::lock_guard <std::mutex> lock (mMutex);
+    std::lock_guard <std::mutex> lock (mutex_);
 
     auto& s = emplace(key).first;
 
-    if ((s.getFlags () & flag) == flag)
+    if (!s.prepRelay(suppressionMap_.clock().now(), holdTime_))
         return false;
 
-    s.swapSet (peers);
-    s.setFlags (flag);
+    s.swap (peers);
 
     return true;
 }
 
+bool HashRouter::prepRelay (uint256 const& key)
+{
+    std::lock_guard <std::mutex> lock (mutex_);
+
+    auto& s = emplace(key).first;
+
+    return s.prepRelay(suppressionMap_.clock().now(), holdTime_);
+}
```

Comment:
There are now two functions named `prepRelay` which do different things. The other one will (as called) always cause `Entry::peers` to become empty; this one will leave it untouched. This is a bit confusing.

Can you explain when one would choose one method over the other, and why the naming makes sense?